vochain/process.go

```diff
@@ -453,18 +453,25 @@ func (app *BaseApplication) NewProcessTxCheck(vtx *models.Tx, txBytes,
 	if acc.Nonce != tx.Nonce {
 		return nil, common.Address{}, ErrAccountNonceInvalid
 	}
-
 	isOracle, err := state.IsOracle(*addr)
 	if err != nil {
 		return nil, common.Address{}, err
 	}
-
-	// check if process entityID matches tx sender
+	// check if process entityID matches tx sender, if not
+	// check if recovered address is an Oracle and if not
+	// check if recovered address is a delegate of the entityID provided
 	if !bytes.Equal(tx.Process.EntityId, addr.Bytes()) && !isOracle {
-		// Check if the transaction comes from an oracle
-		// Oracles can create processes with any entityID
-		return nil, common.Address{}, fmt.Errorf(
-			"unauthorized to create a new process, recovered addr is %s", addr.Hex())
+		entityIDAddress := common.BytesToAddress(tx.Process.EntityId)
+		entityIDAccount, err := state.GetAccount(entityIDAddress, true)
```

Comment:
Are we sure the account should be taken from the last comited state? Please review this (I'm not sure either, but in any case a comment should be addressed to let the decision clear)

Reply:
In this case IMO it make sense. We want to know if an account is a delegate and just in the case that the transaction for adding a delegate and the transaction that allow someone to act as a delegate, if in the list, happen in the same block and are valid will make sense, because they are going to use the volatile state.
Said so it always feels more easy to handle/secure to wait at least to the next block to act as a delegate when added and I cannot imagine a situation in which can be a problem.